base/io.jl

```diff
@@ -884,9 +884,9 @@ end
 
 function read(io::IO, ::Type{Char})
     b0 = read(io, UInt8)::UInt8
-    l = 8(4-leading_ones(b0))
+    l = 0x08 * (0x04 - UInt8(leading_ones(b0)))
```

Comment:
```suggestion
    l = 0x08 * (0x04 -(leading_ones(b0) % UInt8))
```
though maybe LLVM is smart enough to elide the overflow check?

Reply:
```jl
julia> leading_ones_u8(b::UInt8) = UInt8(leading_ones(b))
leading_ones_u8 (generic function with 1 method)

julia> @code_native debuginfo=:none leading_ones_u8(0xff)
	.section	__TEXT,__text,regular,pure_instructions
	.build_version macos, 13, 0
	.globl	_julia_leading_ones_u8_198      ## -- Begin function julia_leading_ones_u8_198
	.p2align	4, 0x90
_julia_leading_ones_u8_198:             ## @julia_leading_ones_u8_198
	.cfi_startproc
## %bb.0:                               ## %top
	cmpb	$-1, %dil
	je	LBB0_1
## %bb.2:                               ## %cond.false
	notb	%dil
	movzbl	%dil, %eax
	bsrl	%eax, %eax
	xorl	$7, %eax
                                        ## kill: def $al killed $al killed $eax
	retq
LBB0_1:
	movb	$8, %al
                                        ## kill: def $al killed $al killed $eax
	retq
	.cfi_endproc
                                        ## -- End function
.subsections_via_symbols
```
Not as concise as I would have imagined, but no error branch and identical to using `% UInt8`.